packages/wrangler/src/r2/catalog.ts

```diff
@@ -0,0 +1,145 @@
+import { createCommand, createNamespace } from "../core/create-command";
+import { logger } from "../logger";
+import { APIError } from "../parse";
+import { requireAuth } from "../user";
+import {
+	disableR2Catalog,
+	enableR2Catalog,
+	getR2Catalog,
+	listR2Catalog,
+} from "./helpers";
+
+export const r2BucketCatalogNamespace = createNamespace({
+	metadata: {
+		description: "Manage R2 bucket warehouses using the R2 Data Catalog",
+		status: "open-beta",
+		owner: "Product: R2 Data Catalog",
+	},
+});
+
+export const r2BucketCatalogEnableCommand = createCommand({
+	metadata: {
+		description: "Enable an R2 bucket as an Iceberg warehouse",
+		status: "open-beta",
+		owner: "Product: R2 Data Catalog",
+	},
+	positionalArgs: ["bucket"],
+	args: {
+		bucket: {
+			describe: "The name of the bucket to enable",
+			type: "string",
+			demandOption: true,
+		},
+	},
+	async handler(args, { config }) {
+		const accountId = await requireAuth(config);
+
+		const response = await enableR2Catalog(accountId, args.bucket);
+
+		logger.log(
+			`✨ Successfully enabled R2 bucket '${args.bucket}' as an Iceberg warehouse. Warehouse name: '${response.name}', id: '${response.id}'.`
+		);
+	},
+});
+
+export const r2BucketCatalogDisableCommand = createCommand({
+	metadata: {
+		description: "Disable R2 bucket as an Iceberg warehouse",
+		status: "open-beta",
+		owner: "Product: R2 Data Catalog",
+	},
+	positionalArgs: ["bucket"],
+	args: {
+		bucket: {
+			describe: "The name of the bucket to disable",
+			type: "string",
+			demandOption: true,
+		},
+	},
+	async handler(args, { config }) {
+		const accountId = await requireAuth(config);
+
+		await disableR2Catalog(accountId, args.bucket);
+
+		logger.log(
+			`✨ Successfully disabled R2 bucket '${args.bucket}' as an Iceberg warehouse.`
+		);
+	},
+});
+
+export const r2BucketCatalogGetCommand = createCommand({
+	metadata: {
+		description: "Check the status of the Iceberg warehouse on an R2 bucket",
+		status: "open-beta",
+		owner: "Product: R2 Data Catalog",
+	},
+	positionalArgs: ["bucket"],
+	args: {
+		bucket: {
+			describe: "The name of the bucket to check",
+			type: "string",
+			demandOption: true,
+		},
+	},
+	async handler(args, { config }) {
+		const accountId = await requireAuth(config);
+
+		try {
+			logger.log(`Fetching warehouse status ...`);
+			const warehouse = await getR2Catalog(accountId, args.bucket);
+			logger.table([
+				{
+					id: warehouse.id,
+					name: warehouse.name,
+					bucket: warehouse.bucket,
+					status: warehouse.status,
+				},
+			]);
+		} catch (e) {
+			if (e instanceof APIError && "not found" in e) {
```

Comment:
"not found" in e seems like a pretty loose condition. Is there an error code or something you can check here?

Reply:
Can check for a 404 instead.